src/renderers/dom/shared/DOMPropertyOperations.js

```diff
@@ -158,6 +158,12 @@ var DOMPropertyOperations = {
         } else if (propertyInfo.hasBooleanValue ||
                    (propertyInfo.hasOverloadedBooleanValue && value === true)) {
           node.setAttribute(attributeName, '');
+        } else if (attributeName === 'value' && node.hasAttribute('value')) {
+          // Use loose coercion to prevent replacement on comparisons like
+          // '3e1' == 30 in Chrome (~52).
+          if (node.value != value) { // eslint-disable-line
+            node.setAttribute(attributeName, '' + value);
```

Comment:
There is an extra complication here... Assigning the `value` attribute forces validation for number inputs. Check this out:

![control](https://cloud.githubusercontent.com/assets/590904/18232983/71c54192-72a9-11e6-8f1f-6141e68d8746.gif)

Notice how the decimal gets stripped? My fix focuses on avoiding touching the input as much as possible to avoid unexpected behavior like this. Fortunately, I believe this also conforms to the standard way inputs work.